akka-http2-support/src/main/scala/akka/http/impl/engine/http2/FrameParser.scala

```diff
@@ -49,6 +49,11 @@ class FrameParser(shouldReadPreface: Boolean) extends ByteStringParser[FrameEven
   def parseFrame(tpe: FrameType, flags: ByteFlag, streamId: Int, payload: ByteReader): FrameEvent = {
     // TODO: add @switch? seems non-trivial for now
     tpe match {
+      case GOAWAY ⇒
+        Http2Compliance.requireZeroStreamId(streamId)
```

Comment:
I actually added `onUpstreamFailure` to the demuxer to clarify what I meant, basically now we might re-add `requirePositiveStreamId` (https://github.com/akka/akka-http/pull/528/files#r88233191) and handle these also in the `FrameParser`, if it is fine to handle parser exceptions in the demuxer.